umh-core/pkg/fsm/redpanda/reconcile.go

```diff
@@ -166,11 +167,42 @@ func (r *RedpandaInstance) reconcileExternalChanges(ctx context.Context, service
 	observedStateCtx, cancel := context.WithTimeout(ctx, constants.RedpandaUpdateObservedStateTimeout)
 	defer cancel()
 
-	err := r.UpdateObservedStateOfInstance(observedStateCtx, services, tick, loopStartTime)
+	err = r.UpdateObservedStateOfInstance(observedStateCtx, services, tick, loopStartTime)
 	if err != nil {
-		return fmt.Errorf("failed to update observed state: %w", err)
+		return fmt.Errorf("failed to update observed state: %w", err), false
+	}
+
+	currentState := r.baseFSMInstance.GetCurrentFSMState()
+	r.baseFSMInstance.GetLogger().Infof("Current state: %s", currentState)
+	if IsRunningState(currentState) {
```

Comment:
maybe add a comment here what this whole if clause is for:
- if there is a config diff AND redpanda is currently running, then apply it via HTTP. because we apply it via HTTP, we can only do it if redpanda is running

Reply:
on the first look, i cannot see how the restart is done? maybe put this here into the comment as well

Reply:
ah, justn oticed that we dont need the restart anymore. Like I said further below, it makes sense to explain all of that reasoning here in docstrings and comments.